nexus/src/app/instance.rs

```diff
@@ -1583,28 +1580,22 @@ impl super::Nexus {
                 | InstanceState::Starting
                 | InstanceState::Stopping
                 | InstanceState::Stopped
-                | InstanceState::Failed => Err(Error::ServiceUnavailable {
-                    internal_message: format!(
-                        "cannot connect to serial console of instance in state \
-                            {:?}",
-                        vmm.runtime.state.0
-                    ),
-                }),
-                InstanceState::Destroyed => Err(Error::ServiceUnavailable {
-                    internal_message: format!(
-                        "cannot connect to serial console of instance in state \
-                        {:?}",
-                        InstanceState::Stopped),
-                }),
+                | InstanceState::Failed => {
+                    Err(Error::invalid_request(format!(
+                        "cannot connect to serial console of {} instance",
```

Comment:
nitty, but I like the previous format a little better.  "cannot connect to serial console of instance in state \"Starting\"" communicates clearly that "Starting" is a discrete state that the user might know about and could go look up what it means and when it would change and how to check for it, etc.  "cannot connect to serial console of starting instance" makes me thing "what do you consider starting?"

Reply:
Sure, I wanted to align this with `cannot connect to serial console of destroyed instance` below, but I think your reasoning makes sense. As a compromise I ended up using the `Display` impl rather than the `Debug` one, so it says `cannot connect to serial console of instance in state "starting"`. (Not wedded to any particular outcome here at all, to be clear.)